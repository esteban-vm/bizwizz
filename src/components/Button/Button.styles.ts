import clsx from 'clsx'

const styles = (isFull?: boolean) =>
  clsx(
    'block min-w-[20rem] cursor-pointer select-none',
    'bg-gradient-120 from-transparent from-0% via-transparent via-50% to-50% bg-[length:250%]',
    'p-[1.6rem] text-2xl font-[500] uppercase transition-all duration-300',
    'hover:translate-x-4 hover:bg-[100%]',
    isFull ? 'bg-primary to-white text-white hover:text-primary' : 'bg-white to-primary text-primary hover:text-white',
    isFull && 'rounded-xl border border-solid border-primary text-center'
  )

export default styles
