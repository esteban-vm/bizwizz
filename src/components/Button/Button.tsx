import type { FC } from '@/types'
import styles from './Button.styles'

type ButtonProps = {
  text: string
}

const Button: FC<ButtonProps> = ({ text }) => {
  return (
    <button className={styles} type='button'>
      {text}
    </button>
  )
}

export default Button
