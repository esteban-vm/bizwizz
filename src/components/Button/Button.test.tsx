import { render, cleanup, screen, userEvent } from '@/tests'
import Button from './Button'

describe('Button:', () => {
  afterEach(cleanup)

  it('should render correctly when type is button', () => {
    const { asFragment } = render(<Button text='click here' />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should call onClick when type is button', async () => {
    const onClick = vi.fn()
    render(<Button onClick={onClick} text='test' />)
    await userEvent.click(screen.getByRole('button'))
    expect(onClick).toHaveBeenCalled()
  })

  it('should render correctly when type is submit', () => {
    const { asFragment } = render(<Button text='submit form' isSubmit />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should not call onClick when type is submit', async () => {
    const onClick = vi.fn()
    render(<Button onClick={onClick} text='test' isSubmit />)
    await userEvent.click(screen.getByRole('button'))
    expect(onClick).not.toHaveBeenCalled()
  })

  it('should have correct structure', () => {
    render(<Button text='test' />)
    const buttonElem = screen.getByRole('button')
    expect(buttonElem).toBeInTheDocument()
    expect(buttonElem).toHaveTextContent(/test/i)
  })
})
