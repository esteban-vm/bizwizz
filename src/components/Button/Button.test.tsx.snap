// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button: > should render correctly when type is button 1`] = `
<DocumentFragment>
  <button
    class="block min-w-[20rem] cursor-pointer select-none bg-gradient-120 from-transparent from-0% via-transparent via-50% to-50% bg-[length:250%] p-[1.6rem] text-2xl font-[500] uppercase transition-all duration-300 hover:translate-x-4 hover:bg-[100%] bg-white to-primary text-primary hover:text-white"
    type="button"
  >
    click here
  </button>
</DocumentFragment>
`;

exports[`Button: > should render correctly when type is submit 1`] = `
<DocumentFragment>
  <button
    class="block min-w-[20rem] cursor-pointer select-none bg-gradient-120 from-transparent from-0% via-transparent via-50% to-50% bg-[length:250%] p-[1.6rem] text-2xl font-[500] uppercase transition-all duration-300 hover:translate-x-4 hover:bg-[100%] rounded-xl border border-solid border-primary bg-primary to-white text-white hover:text-primary"
    type="submit"
  >
    submit form
  </button>
</DocumentFragment>
`;
